javav2/usecases/creating_fsa_app/src/main/java/com/example/fsa/services/S3Service.java

```diff
@@ -0,0 +1,46 @@
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+package com.example.fsa.services;
+
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class S3Service {
+
+    // Put the audio file into the Amazon S3 bucket.
+    public String putAudio(InputStream is, String bucket, String key) throws IOException {
+        S3Client s3 = S3Client.builder()
+            .region(Region.US_EAST_1)
+            .build();
+
+        PutObjectRequest putOb = PutObjectRequest.builder()
+            .bucket(bucket)
+            .contentType("audio/mp3")
+            .key(key)
+            .build();
+
+        s3.putObject(putOb, RequestBody.fromBytes(inputStreamToBytes(is)));
```

Comment:
Does `RequestBody.fromInputStream` not work here?
https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/sync/RequestBody.html#fromInputStream(java.io.InputStream,long)